ui/ui-components/pages/app/[id]/refresh.tsx

```diff
@@ -45,7 +46,8 @@ export default function Page(props) {
   );
   const { id } = router.query;
   const { schedules, runs, sources } = props;
-  const disabled = app?.locked !== null || appRefreshQuery?.locked !== null;
+  const disabled =
+    app?.locked?.length > 1 || appRefreshQuery?.locked?.length > 1;
```

Comment:
Was this part of the code still causing problems?  Was there still a `undefined` somewhere?

Reply:
I'd checkout this part of the code for maying making the thing undefined:

```
    setAppRefreshQuery({
      ...appRefreshQuery,
      [event.target.id]: event.target.value,
```

Reply:
yep. I'll look there.